tasks/nunjucks.js

```diff
@@ -35,7 +35,7 @@ module.exports = function(grunt) {
 
         async.each(this.files, function(f, done) {
             var filepath = path.join(process.cwd(), f.src[0]);
-            var data = options.data || {};
+            var data = JSON.parse(JSON.stringify(options.data || {}));
```

Comment:
@ArmorDarks Seems I've figured out what's happening.

Nunjucks adds all variables declared with `set` to the data object which was specified for `render` function.

As long as all **objects in javascript passes to functions by reference** the next template gets already modified version of data, so there's variable declared in the first template.

Cloning data object each time before passing it to the `render` function solves the issue.